### DISCUSSION TITLE
Legend not showing when values are `var(...)`

### DISCUSSION THREAD

**Discussion by vincent-octo** [observablehq/plot#2013] [Q&A] [answered]
Hello,

When using `color: {legend: true}` and values are `"var(--something)"` then the legend does not appear.
Additionally, I could not see any warning or error message that could help in understanding why the legend doesn't show.

Here is an notebook that reproduces the bug: https://observablehq.com/d/fbd79535abd8788a
Changing `fill: "test"` with `fill: "sex"` will display the legend again, so I am assuming that the `"var(...)"` content of `test` is to blame here.

**Comment by Fil** [comment, marked as answer]
When the values are passed as literal colors, there is no color scale, and as a consequence no legend.

to fix, you could try using "sex" again, with something like this for the color option:
```
color: {
  legend: true,
  domain: ["FEMALE", "MALE", null],
  range: ["var(--female)", "var(--male)", "var(--null)"]
},
```

**Reply by vincent-octo** [reply]
Thanks! That does indeed fix it. :)

I was a little lost as to why the bug occured at first, and a hint from an error/warning message would have probably saved some time. Could adding such a message be considered?

**Reply by Fil** [reply]
I've added this to the list in https://github.com/observablehq/plot/issues/755
